contracts/UniswapV3Staker.sol

```diff
@@ -292,22 +295,32 @@ contract UniswapV3Staker is
             SafeMath.sub(incentives[incentiveId].totalRewardUnclaimed, reward)
         );
 
-        /* TODO: This will be fixed in https://github.com/omarish/uniswap-v3-staker/issues/38
-        so that collecting and unstaking are two separate functions */
-        try
-            IERC20Minimal(incentives[incentiveId].rewardToken).transfer(
-                params.to,
-                reward
-            )
-        returns (bool) {
-            // It didn't fail
-        } catch {
-            // It failed
-        }
+        rewards[incentives[incentiveId].rewardToken][msg.sender] = uint128(
+            SafeMath.add(rewards[incentives[incentiveId].rewardToken][msg.sender], reward)
+        );
 
         emit TokenUnstaked(params.tokenId);
     }
 
+    /// @inheritdoc IUniswapV3Staker
+    function claimReward(address rewardToken) external override {
+        uint128 reward = rewards[rewardToken][msg.sender];
+
+        require(reward != 0, 'NO_REWARDS_AVAILABLE');
```

Comment:
Maybe this is opinionated but I'm a fan of taking out this check. It costs gas, and transferring 0 tokens is not harmful. for instance, openzeppelin does not check that amount is 0 on erc20 transfers.

Reply:
I think that's fair, going to remove. What do you think about wrapping this in an if statement instead? Something along the lines of this? Happy with either implementation (the one you suggested above).

```
        if (reward > 0) {
            rewards[rewardToken][msg.sender] = 0; 
            require(
                IERC20Minimal(rewardToken).transfer(
                    msg.sender,
                    reward
                ),
                'REWARD_TRANSFER_FAILED'
            );
            emit RewardClaimed(msg.sender, reward);
        }
```